servicemanager/servicemanager.go

```diff
@@ -342,3 +548,114 @@ func (sm *ServiceManager) prepareServiceFS(imagePath string, gid int) (rootFSDig
 
 	return rootFSDigest, nil
 }
+
+func (sm *ServiceManager) removeDamagedServiceFolders(services []ServiceInfo) error {
+	files, err := ioutil.ReadDir(sm.servicesDir)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+filesLoop:
+	for _, file := range files {
+		fullPath := path.Join(sm.servicesDir, file.Name())
+
+		for _, service := range services {
+			if fullPath == service.ImagePath {
+				continue filesLoop
+			}
+		}
+
+		if err = os.RemoveAll(fullPath); err != nil {
+			log.Errorf("Can't remove service directory: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (sm *ServiceManager) tryAllocateSpace(requiredSize int64) (err error) {
+	sm.Lock()
+	defer sm.Unlock()
+
+	if sm.numTryAllocateSpace == 0 {
+		if sm.availableSize, err = GetAvailableSize(sm.servicesDir); err != nil {
+			return aoserrors.Wrap(err)
+		}
+	}
+
+	for sm.availableSize < requiredSize {
+		if len(sm.pendingRemove) == 0 {
+			return aoserrors.New("not enough memory to delete service")
```

Comment:
error message sounds weird: just `not enough memory`.

Reply:
Done